src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -140,26 +148,32 @@ export class ItemDisplayComponent implements OnInit, AfterViewInit, OnDestroy {
 }
 
 export class ItemDisplayPlatform extends Platform {
-  taskParams: any;
+  taskParams: TaskParams;
   constructor(task: Task, taskParams: TaskParams) {
     super(task);
     this.taskParams = taskParams;
   }
 
-  validate(mode : string, success : CompleteFunction, _error : ErrorFunction) : void {
+  validate(mode : string) : Observable<void> {
     if (mode == 'cancel') {
       // TODO reload answer
+      return of();
     }
     if (mode == 'validate') {
-      this.task.getAnswer((answer: string) => {
-        this.task.gradeAnswer(answer, '', (results : any) => {
-          success(results);
-        });
-      });
+      return this.task.getAnswer()
+        .pipe(
+          switchMap((answer: string) => this.task.gradeAnswer(answer, '')),
```

Comment:
as the observable is typed, it is useless to repeat the type
```suggestion
          switchMap(answer => this.task.gradeAnswer(answer, '')),
```